gwy/templates/go.mod.tmpl

```diff
@@ -0,0 +1,7 @@
+module gateway
+
+go 1.14
+
+require (
+	github.com/grpc-ecosystem/grpc-gateway/v2 v2.0.1
```

Comment:
I think we can use this file as an opportunity to pin all the current modules that are getting pulled.
I'm guessing that when we build the binary it updates the go mod and go.sum file, which btw, I think we should also add to the template.

Reply:
> I think we can use this file as an opportunity to pin all the current modules that are getting pulled.

What do you mean? Any examples?

Reply:
yes the `go.sum` file will be created and `go.mod` updated when we build. I agree on pinning the other dependencies as well.
@vtopc the gwy gateway file has other dependencies that could use pinning. see [this](https://github.com/namely/docker-protoc/blob/86f2ef38537cd3475c898f436655819f19e9b354/gwy/templates/main.go.tmpl#L16-L23)